src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -229,6 +230,21 @@ def get_task_res(self, task_ids: set[UUID]) -> list[TaskRes]:
                 current_time=current,
             )
 
+            # Check node availability
```

Comment:
I'd suggest let's move this functionality to `get_message_res`. The utility function `check_node_availability_for_taskins` will need to be slightly updated so it operates on `Messages` only